### PR TITLE
Added allocations to schedulers

### PR DESCRIPTION
**Pull request by jshaofuturewei**
Added allocations to schedulers

Run "hack/globalscheduler/globalscheduler-up.sh"
Run "cluster/kubectl.sh apply -f globalscheduler/test/yaml/sample_2_schedulers.yaml"
Run "cluster/kubectl.sh apply -f globalscheduler/test/yaml/sample_4_clusters.yaml"
Run "cluster/kubectl.sh apply -f globalscheduler/test/yaml/sample_2_distributors.yaml"
Run "cluster/kubectl.sh apply -f globalscheduler/test/yaml/sample_2_allocations.yaml"
Run "cluster/kubectl.sh get allocations" and found the 2 allocations status phases have changed to "Scheduled"

### CODE REVIEW

File: globalscheduler/test/yaml/sample_2_allocations.yaml [hunk 1]
```diff
@@ -71,7 +71,7 @@ spec:
     regions:
       - region: "NE-1"
         availability_zone: ["production-az"]
```
Comment (kimeunju108) [inline globalscheduler/test/yaml/sample_2_allocations.yaml:73]: isn't az "production-az"  instead of ["production-az"]?

Comment (jshaofuturewei) [reply globalscheduler/test/yaml/sample_2_allocations.yaml:73]: According to the allocation spec, availability_zone is an array of string not a string.

Comment (kimeunju108) [reply globalscheduler/test/yaml/sample_2_allocations.yaml:73]: In that case, how to generate key field "SiteID" key because SiteID is Region--AZ.

Comment (jshaofuturewei) [reply globalscheduler/test/yaml/sample_2_allocations.yaml:73]: According to the logs, it is "NE-1--production-az" as we defined in the sample_n_pods.yaml. The allocation is the collection of pods. Therefore it is the same as we defined in pods

Comment (kimeunju108) [reply globalscheduler/test/yaml/sample_2_allocations.yaml:73]: When AZ array size is more than 2, what is the SiteID?  Could you clarify that whether AZ array size is always 1 ?

Comment (jshaofuturewei) [reply globalscheduler/test/yaml/sample_2_allocations.yaml:73]: In the https://github.com/CentaurusInfra/global-resource-scheduler/blob/master/globalscheduler/pkg/scheduler/eventhandlers.go#L259, it picks the first one.

**Review by kimeunju108 — APPROVED** [review]
It looks like there is some alignment needed between data structure (array) and logic (siteid).

**Review by CoderKevinZhang — APPROVED** [review]

**Pull request merged**